catalyst_voices/packages/catalyst_voices_blocs/lib/src/brand/brand_bloc.dart

```diff
@@ -1,71 +1,73 @@
 import 'package:bloc/bloc.dart';
-
-import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
 import 'package:equatable/equatable.dart';
 import 'package:flutter/material.dart';
 
 part 'brand_event.dart';
 part 'brand_state.dart';
 
-// [BrandBloc] is a Bloc responsible for managing the brand state.
-// This Bloc listens for [BrandEvent]s and updates the [BrandState]
-// accordingly.
-// The [BrandState] can be passed to the [MaterialApp] as a theme
-// allowing to change the theme at runtime.
-//
-// The [BrandChanged] accepts a [BrandKey] and uses that is used for
-// the selection of the related [ThemeData] defined in the [brands]
-// map.
-// To trigger the theme change is just necessary to dispatch the
-// [BrandChanged] with the appropriate key:
-//
-// ```dart
-// context.read<BrandBloc>().add(
-//   const BrandChanged(BrandKey.catalyst),
-// );
-// ```
-//
-// Usage example:
-//
-// ```dart
-// BlocProvider(
-//   create: (context) => BrandBloc(),
-//   child: BlocBuilder<BrandBloc, BrandState>(
-//     builder: (context, state) {
-//       return MaterialApp(
-//         builder: (context, state) => Scaffold(
-//           body: Row(
-//             children: [
-//               MaterialButton(
-//                 color: Theme.of(context).primaryColor,
-//                 onPressed: () {
-//                   context.read<BrandBloc>().add(
-//                     const BrandChanged(BrandKey.catalyst),
-//                   );
-//                 },
-//                 child: const Text('Switch to Catalyst Theme'),
-//               ),
-//             ],
-//           ),
-//         ),
-//         theme: state.themeData,
-//       );
-//     },
-//   ),
-// );
-// ```
-
-
+/// [BrandBloc] is a Bloc responsible for managing the brand state.
+/// 
+/// This Bloc listens for [BrandEvent]s and updates the [BrandState]
+/// accordingly.
+/// The [BrandState] can be consumed by the [MaterialApp] to build the 
+/// appropriate [ThemeData] based on the [BrandKey] stored in the state.
+/// 
+/// To build the appropriate [ThemeData] based on the [BrandKey] is
+/// possible to use the [ThemeBuilder] utility form the `catalyst_voices_brands`
+/// package.
+///
+/// The [BrandChangedEvent] accepts a [BrandKey] a simple enum that
+/// contains all possible brands/themes.
+/// 
+/// To trigger the theme change is just necessary to dispatch the
+/// [BrandChangedEvent] with the appropriate key:
+///
+/// ```dart
+/// context.read<BrandBloc>().add(
+///   const BrandChangedEvent(BrandKey.catalyst),
+/// );
+/// ```
+///
+/// Usage example:
+///
+/// ```dart
+/// BlocProvider(
+///   create: (context) => BrandBloc(),
+///   child: BlocBuilder<BrandBloc, BrandState>(
+///     builder: (context, state) {
+///       return MaterialApp(
+///         builder: (context, state) => Scaffold(
```

Comment:
`state` param should be called `child`, but - do we need the builder anyway? Just put the scaffold under `home` property.

Reply:
Changed the param name. The `context` is needed to extract the primary color from theme. As suggested I used `home` and a Builder to expose proper context.